### PR TITLE
Feature: FAB

### PR DESCRIPTION
**Pull request by NicolasChampagne**
GitHub Issue: #

## PR Type

What kind of change does this PR introduce?

- Feature

## Description

- Fab for icons and text+icons
- Known issue for wasm ElevatedView and Ellipse Fill with opacity

## PR Checklist 
Please check if your PR fulfills the following requirements:

- Commits must be following the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] Tested UWP Latest Upgrade to uwp 18, causes build issue. Works on uwp16 and 17
- [x] Tested iOS
- [x] Tested Android
- [x] Tested WASM
- [x] Contains **No** breaking changes
  > If the pull request contains breaking changes, commit message must contain a detailed description of the action to take for the consumer of this library. As explained by the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: src/samples/Uno.Material.Samples/Uno.Material.Samples.Shared/Content/Controls/FabSamplePage.xaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			<Path Data="M19 13 13 13 13 19 11 19 11 13 5 13 5 11 11 11 11 5 13 5 13 11 19 11z"
+				  Fill="Black"
+				  HorizontalAlignment="Center"
+				  VerticalAlignment="Center"
+				  Margin="0,0,6,4" />
+		</win:Button>
+
+		<!-- Known Issue, ElevatedView adds white border around fab in WASM -->
+		<not_win:Grid>
+			<toolkit:ElevatedView Elevation="6"
```
Comment (MatFillion) [inline src/samples/Uno.Material.Samples/Uno.Material.Samples.Shared/Content/Controls/FabSamplePage.xaml:34]: I would prefer the elevatedView inside the Button's Style (in the library). Does it make sense?

Comment (NicolasChampagne) [reply src/samples/Uno.Material.Samples/Uno.Material.Samples.Shared/Content/Controls/FabSamplePage.xaml:34]: I can't apply it there... using toolkit in the library still causes a compile error

Comment (MatFillion) [reply src/samples/Uno.Material.Samples/Uno.Material.Samples.Shared/Content/Controls/FabSamplePage.xaml:36]: @NicolasChampagne  PR #46 will solve this!

File: src/library/Uno.Material/Styles/Controls/Button.xaml [hunk 2]
```diff
@@ -33,13 +34,23 @@
...
+	<!-- Floating Action Button Variables -->
+	<SolidColorBrush x:Key="FabBackgroundColor"
+					 Color="{StaticResource SecondaryColor}" />
+	<SolidColorBrush x:Key="FabIconColor"
+					 Color="{StaticResource OnSurfaceColor}" />
+	<SolidColorBrush x:Key="FabLabelColor"
+					 Color="{StaticResource OnSurfaceColor}" />
+	<CornerRadius x:Key="FabBorderRadius">23</CornerRadius>
+	<x:Double x:Key="FabLargeAreaSize">56</x:Double>
+	<x:Double x:Key="FabSmallAreaSize">40</x:Double>
```
Comment (MatFillion) [inline src/library/Uno.Material/Styles/Controls/Button.xaml:46]: I think small area is not used?

File: src/library/Uno.Material/Styles/Controls/Button.xaml [hunk 3]
```diff
@@ -33,13 +34,23 @@
...
 					 Color="{ThemeResource PrimaryColor}" />
 
+	<!-- Floating Action Button Variables -->
+	<SolidColorBrush x:Key="FabBackgroundColor"
+					 Color="{StaticResource SecondaryColor}" />
+	<SolidColorBrush x:Key="FabIconColor"
+					 Color="{StaticResource OnSurfaceColor}" />
+	<SolidColorBrush x:Key="FabLabelColor"
+					 Color="{StaticResource OnSurfaceColor}" />
+	<CornerRadius x:Key="FabBorderRadius">23</CornerRadius>
```
Comment (MatFillion) [inline src/library/Uno.Material/Styles/Controls/Button.xaml:44]: seems to be 28 in the zeplin

File: src/library/Uno.Material/Styles/Controls/Button.xaml [hunk 4]
```diff
@@ -33,13 +34,23 @@
...
+	<x:Double x:Key="FabLargeAreaSize">56</x:Double>
```
Comment (MatFillion) [inline src/library/Uno.Material/Styles/Controls/Button.xaml:45]: Is there a way to make it a multiplication of the radius ?
@davidjohnoliver  @agneszitte-nventive @Massimo37  looking for ideas =D
We could do a MultiplyConverter and Binding RelativeSourceTemplatedParent using that converter but it seems quite hackish

File: src/library/Uno.Material/Styles/Controls/Button.xaml [hunk 5]
```diff
@@ -424,4 +435,272 @@
 		</Setter>
 	</Style>
 
+	<!-- Issue #41: Todo remove not_wasm when fix for Ellipse Fill with SolidColorBrushes that contain opacity is merged -->
```
Comment (MatFillion) [inline src/library/Uno.Material/Styles/Controls/Button.xaml:438]: I noticed this style isn't shown in the samples, we have 3 styles but only 2 samples.

File: src/library/Uno.Material/Styles/Controls/Button.xaml (hunk 3 above)
Comment (NicolasChampagne) [reply src/library/Uno.Material/Styles/Controls/Button.xaml:44]: I wrote in the zeplin that in the zeplin their radius is 23, but their written number is 28, 28 is too mch and makes the button look like a disc

File: src/samples/Uno.Material.Samples/Uno.Material.Samples.Shared/Content/Controls/FabSamplePage.xaml (hunk 1 above)
Comment (NicolasChampagne) [reply src/samples/Uno.Material.Samples/Uno.Material.Samples.Shared/Content/Controls/FabSamplePage.xaml:36]: thanks

File: src/library/Uno.Material/Styles/Controls/Button.xaml (hunk 2 above)
Comment (NicolasChampagne) [reply src/library/Uno.Material/Styles/Controls/Button.xaml:46]: fixed

File: src/library/Uno.Material/Styles/Controls/Button.xaml (hunk 4 above)
Comment (NicolasChampagne) [reply src/library/Uno.Material/Styles/Controls/Button.xaml:45]: do I wait on this?

**Comment by NicolasChampagne** [comment]
Visual studio 2019 is reverting my changes every time I push them.... Wasted quite some time on this....

File: src/library/Uno.Material/Styles/Controls/Button.xaml (hunk 4 above)
Comment (MatFillion) [reply src/library/Uno.Material/Styles/Controls/Button.xaml:44]: no need to wait @NicolasChampagne We can improve later.

File: src/library/Uno.Material/Styles/Application/Colors.xaml [hunk 6]
```diff
@@ -72,7 +72,7 @@
 	<!-- Issue #41 TODO: SolidColorBrushes set this way do not function with android thumbtints and tracktints -->
 	<SolidColorBrush x:Key="OnPrimaryBrush"
 					 Color="{StaticResource PrimaryColor}"
-					 Opacity="1.0"/>
+					 Opacity="0.0"/>
```
Comment (MatFillion) [inline src/library/Uno.Material/Styles/Application/Colors.xaml:75]: @NicolasChampagne  why this change
